File: Beelina.APP/src/app/_models/top-supplier-by-sales.ts
export class TopSupplierBySales {
  public supplierId: number;
  public supplierName: string;
  public supplierCode: string;
  public totalSalesAmount: number;
  public totalProductsSold: number;
  public totalTransactions: number;
  public totalSalesAmountFormatted: string;

  constructor(data?: Partial<TopSupplierBySales>) {
    this.supplierId = data?.supplierId || 0;
    this.supplierName = data?.supplierName || '';
    this.supplierCode = data?.supplierCode || '';
    this.totalSalesAmount = data?.totalSalesAmount || 0;
    this.totalProductsSold = data?.totalProductsSold || 0;
    this.totalTransactions = data?.totalTransactions || 0;
    this.totalSalesAmountFormatted = data?.totalSalesAmountFormatted || '';
  }

  get nameWithCode() {
    return `${this.supplierCode} - ${this.supplierName}`;
  }
}

File: Beelina.LIB/Models/TopSupplierBySales.cs
namespace Beelina.LIB.Models
{
    public class TopSupplierBySales
    {
        public int SupplierId { get; set; }
        public string SupplierName { get; set; }
        public string SupplierCode { get; set; }
        public double TotalSalesAmount { get; set; }
        public int TotalProductsSold { get; set; }
        public int TotalTransactions { get; set; }
        public string TotalSalesAmountFormatted
        {
            get
            {
                return TotalSalesAmount.ToString("₱#,##0.00");
            }
        }
    }
}